src/derived/floor.js

```diff
@@ -1,5 +1,6 @@
 import { sub, round } from '../base';
 
 export default function floor(a) {
-  return round(sub(a, 0.5));
+  const b = round(a);
+  return cond(lessThan(a, b), round(sub(a, 0.5)), b);
```

Comment:
I think we don't need it to be this complex. I think `round(sub(a, 0.5 - Number.MIN_VALUE))` should do.

Reply:
This suggested solution doesn't work, what is the strategy you are trying to use?

Reply:
So the problem here is that -0.5 gets rounded to -1 and not to 0. If we subtract fewer than 0.5 from the value we calculate floor for, then it should get rounded in the correct direction. I'd try with sth like `0.5 - 1e-8` if `MIN_VALUE` doesn't work.

Reply:
This solution is not working either

Reply:
I checked the following implementation:
```
function floor(a) {
  return round(sub(a, 0.5 - 1e-8));
}
```

Output:
```
IN 1 OUT 1
IN 0.9 OUT 0
IN 0.1 OUT -0
IN 0 OUT -0
IN -0.1 OUT -1
IN -0.9 OUT -1
IN -1 OUT -1
```

And appears to be working fine. If possible I'd prefer to stick to a smaller number of nodes